### DISCUSSION TITLE
Compatible with WHMCS?

### DISCUSSION THREAD

**Discussion by drumship3** [oidc-wp/openid-connect-generic#525] [Q&A]
I am just verifying if this is compatible with whmcs openid connect? I am continually getting: ERROR (invalid-user-claim): Invalid user claim. I probably have a setting wrong.

**Comment by timnolte** [comment]
Are you using WHMCS as your IDP? I have not tested it personally, however so long as WHMCS supports the OpenID Connect standard there shouldn't be an issue.

**Reply by drumship3** [reply]
yes I am. Thanks for your fast reply! I am sure i have something set wrong as I am new to this!

**Reply by timnolte** [reply]
Yeah, unfortunately I don't have a WHMCS instance available to test with, given it's commercial with no free licensing available. I'm not sure if it would be helpful for you to review some of our IDP setup guides that might give you hints as to the proper configuration on the WHMCS side.

**Comment by timnolte** [comment]
There is mention of something that might work in terms of setting the UserInfo endpoint blank. Have you tried that?

https://github.com/oidc-wp/openid-connect-generic/issues/152

**Reply by drumship3** [reply]
I will try that now....

**Comment by timnolte** [comment]
It might also be helpful to know what scopes you are trying to use and also which claims WHMCS support. There are some details about claims here for your information:

https://connect2id.com/products/server/docs/guides/requesting-openid-claims

**Comment by drumship3** [comment]
no, it still isn't working it returned: ERROR (no-username): No appropriate username found.

OpenIDScope: email profile openid

**Reply by timnolte** [reply]
OK, well that is a different error, so it's not running into the same issue. How do you have the plugin settings setup in terms of username to use from your IDP and whether to create a new user if it doesn't exist or can't find a match?

**Reply by drumship3** [reply]
i have it checked to create user if doesn't exist....

**Reply by drumship3** [reply]
what setting is this?: username to use from your IDP

**Comment by timnolte** [comment]
Have you taken a look at the `.well-known` OpenID Connect configuration endpoint to see what it says are the settings to use? I'm looking at the documents here and it provides examples of the Authorization Endpoint.

https://docs.whmcs.com/OpenID_Connect_Developer_Guide

**Reply by drumship3** [reply]
no i have not...i will look at it now...

**Reply by drumship3** [reply]
i did not notice any change in behavior. I added this: htaccess in public_html root and changed the path to /clients where whmcs is installed.

# OpenID Discovery Document (http://openid.net/specs/openid-connect-discovery-1_0.html)
RewriteRule ^.well-known/openid-configuration ./clients/oauth/openid-configuration.php [L,NC]

**Reply by timnolte** [reply]
So what I'm asking is not about change in behavior. You should be able to load your `/.well-known/openid-configuration` endpoint in your browser and see all of the proper configuration items to use. For example on one of my test IDPs I have this endpoint which you can access and see what sort of response is returned.

https://keycloak.oidcwp.net/auth/realms/oidcwp-net/.well-known/openid-configuration

**Reply by drumship3** [reply]
yea ok. excellent! I will look closely at this!
